benchmarks/priority_queue/priority_queue_bench.cu

```diff
@@ -0,0 +1,87 @@
+#include <cstdint>
+#include <random>
+#include <vector>
+
+#include <benchmark/benchmark.h>
+
+#include <cuco/detail/pair.cuh>
+#include <cuco/priority_queue.cuh>
+
+#include <thrust/device_vector.h>
+
+using namespace cuco;
+
+template <typename T>
+struct pair_less {
+  __host__ __device__ bool operator()(const T& a, const T& b) const { return a.first < b.first; }
+};
+
+template <typename Key, typename Value, typename OutputIt>
+static void generate_keys_uniform(OutputIt output_begin, OutputIt output_end)
+{
+  std::random_device rd;
+  std::mt19937 gen{rd()};
+
+  auto num_keys = std::distance(output_begin, output_end);
+
+  for (auto i = 0; i < num_keys; ++i) {
+    output_begin[i] = {static_cast<Key>(gen()), static_cast<Value>(gen())};
+  }
+}
```

Comment:
Can this be replaced by `generate_keys` in `benchmarks/key_generator.hpp`?

Reply:
This won't work directly, as this function is generating key-value pairs rather than just keys. We could use `generate_keys` to generate both keys and values and then combine them or use it to generate keys and then assign sequential values though. For now, I have renamed `generate_keys_uniform` -> `generate_kv_pairs_uniform` for clarity (aa12404).